lib/doh.c

```diff
@@ -425,8 +425,9 @@ Curl_addrinfo *Curl_doh(struct connectdata *conn,
   error:
   curl_slist_free_all(data->req.doh.headers);
   data->req.doh.headers = NULL;
-  Curl_close(&data->req.doh.probe[0].easy);
-  Curl_close(&data->req.doh.probe[1].easy);
+  for(int slot = 0; slot < DOH_PROBE_SLOTS; slot++) {
```

Comment:
This is not C89 compliant. The variable needs to be declared at the beginning on the block.

Reply:
Thanks.